core/src/main/java/com/linecorp/armeria/common/multipart/Multipart.java

```diff
@@ -432,4 +433,16 @@ default <T> CompletableFuture<List<T>> collect(
             SubscriptionOption... options) {
         return bodyParts().mapAsync(function).collect(options);
     }
+
+    /**
+     * Make only those that meet certain conditions in a body part a multipart.
+     *
+     * @param predicate certain conditions
+     * @return multipart matching the condition
+     */
+    default Multipart filterBodyParts(Predicate<BodyPart> predicate) {
```

Comment:
I think it's more intuitive to stick with names often used in fp

```suggestion
    default Multipart filter(Predicate<BodyPart> predicate) {
```